osu.Game.Rulesets.Osu/Difficulty/OsuLegacyScoreSimulator.cs

```diff
@@ -133,22 +129,25 @@ private void simulateHit(HitObject hitObject)
                     // The spinner object applies a lenience because gameplay mechanics differ from osu-stable.
                     // We'll redo the calculations to match osu-stable here...
                     const double maximum_rotations_per_second = 477.0 / 60;
-                    double minimumRotationsPerSecond = IBeatmapDifficultyInfo.DifficultyRange(playableBeatmap.Difficulty.OverallDifficulty, 3, 5, 7.5);
+
+                    // Normally, this value depends on the final overall difficulty. For simplicity, we'll only consider the worst case that minimises bonus score.
+                    const double minimum_rotations_per_second = 7.5;
```

Comment:
I'm trying to make heads and tails of this and I'm not sure it's making sense to me. The thought process is as follows:

- We are assuming the maximum... minimum of number of rotations per second here, OD-wise. This means that the rotation requirement is going to be as high as it can possibly can go here.
- Therefore, setting this to max also maximises `halfSpinsRequiredForCompletion`.
- That indeed does mean that the total final achievable legacy bonus score for spinners will be _minimised_.
- However, after this I'm lost, because what happens later is that `BonusScoreRatio` is computed as `standardisedBonusScore / legacyBonusScore`. If the denominator is _minimised_, the quotient itself is _maximised_.
- The score migrator uses `BonusScoreRatio` by taking the legacy total score, by taking the total score of a given stable score, subtracting the max legacy base score, and if there is any excess left, then that is considered bonus and is multiplied by `BonusScoreRatio`. But as stated above, the ratio is _maximised_ here, and the OP says:

> where the number of normal ticks is maximised (bonus minimised), so that every score set can theoretically be beaten assuming enough bonus.

and is where I get lost, because it seems to me that this _benefits_ legacy scores...? Where am I wrong?

Reply:
The above does seem to check out.

Reply:
Hmmm... I guess one of the biggest issues is that I thought normal ticks were part of `AccuracyScore`, but this isn't the case - they're actually bonus.

Excluding that, my thought process here (if normal ticks were part of `AccuracyScore`) was to maximise `(AccuracyScore + ComboScore)` thereby minimising the calculation of the non-bonus score part when converting high scores.  
In that scenario, if the only bonus was from `SpinnerBonusTick`, `BonusScoreRatio` is actually a straight up constant equal to `50 / 1100` so there's no such maximisation/minimisation of the denominator.  

But since normal ticks are bonus, then I believe it is true that we want to minimise the number of normal ticks as they're the dominator of the equation.

We may also want to consider removing `BonusScoreRatio` and treating it as a constant of `0.045`.

Reply:
Right, everything you said above checks out to me :+1: